components/select/select.component.ts

```diff
@@ -400,6 +400,10 @@ export class NzSelectComponent implements ControlValueAccessor, OnInit, AfterCon
     const activatedIndex = listOfFilteredOptionNotDisabled.findIndex(item =>
       this.compareWith(item.nzValue, this.activatedValue)
     );
+    // check option value is exist
+    if (!listOfFilteredOptionNotDisabled.length) {
+      return;
+    }
```

Comment:
This logic should not be placed here, it only affects the actions of `UP_ARROW` `DOWN_ARROW` `ENTER`.
We don't have enough tests here, it would be better if you added tests here.

Reply:
ok

Reply:
The action of  `ENTER` does not need to change,
And has add a test case for this situation